```{r libraries}
library(tidyverse)
library(cowplot)
library(ggVennDiagram)
library(tidyverse)
library(glue)
library(ggtext)
library(ggrepel)
library(sjmisc) # for rotate_df
```

# Read input data
## Assigned and unassigne Orthogroups
```{r}
## assigend and unassinged orthogroups or EV, EG and Musa genomes 


EV_AED25_TE.OG_GeneCount <-
  read.delim("orthofinder/Orthogroups/Orthogroups.GeneCount.tsv", sep = '\t', header = T) %>%  
  mutate(
    mazia = case_when(mazia >=1 ~ 1,
                                         TRUE ~ 0),
    bedadeti = case_when(bedadeti >=1 ~ 1,
                                            TRUE ~ 0)
    # Musa_acuminata = case_when(Musa_acuminata >=1 ~ 1,
    #                            TRUE ~ 0),
    # Musa_balbisiana = case_when(Musa_balbisiana >=1 ~ 1,
    #                             TRUE ~ 0)
    # Musa_schizocarpa = case_when(Musa_schizocarpa >=1 ~ 1,
    #                              TRUE ~ 0)
  ) %>%
  
  ## add unassigned OGs 
  bind_rows(
    read.delim("orthofinder/Orthogroups/Orthogroups_UnassignedGenes_TE.tsv", header = T, sep = '\t') %>%
      mutate(
        mazia = case_when(str_detect(mazia,"\\w+") ~ 1,
                                             TRUE ~ 0),
        bedadeti = case_when(str_detect(bedadeti, "\\w+") ~ 1,
                                                TRUE ~ 0)
        # Musa_acuminata = case_when(str_detect(Musa_acuminata, "\\w+") ~ 1,
        #                            TRUE ~ 0),
        # Musa_balbisiana = case_when(str_detect(Musa_balbisiana, "\\w+") ~ 1,
        #                             TRUE ~ 0)
        # Musa_schizocarpa = case_when(str_detect(Musa_schizocarpa, "\\w+") ~ 1,
        #                              TRUE ~ 0)
      )) %>% select(-Total)


```  



## Prepare dat for Ven Diagram
```{r}
dir.create("orthofinder/Orthogroups/VennDiag/")


EV_AED25_TE.OG_GeneCount %>%
  filter(mazia == 1) %>% 
  select(Orthogroup) %>%
  write.table("orthofinder/Orthogroups/VennDiag/mazia.OG.txt", col.names = F, sep = '\t', quote = F, row.names = F)

EV_AED25_TE.OG_GeneCount %>%
  filter(bedadeti == 1) %>% 
  select(Orthogroup) %>% 
  write.table("orthofinder/Orthogroups/VennDiag/bedadeti.OG.txt", col.names = F, sep = '\t', quote = F, row.names = F)


# EV_AED25_TE.OG_GeneCount %>%
#   filter(Musa_acuminata == 1) %>% 
#   select(Orthogroup) %>% 
#   write.table("orthofinder/Orthogroups/VennDiag/Musa_acuminata.OG.txt", col.names = F, sep = '\t', quote = F, row.names = F)
# 
# EV_AED25_TE.OG_GeneCount %>%
#   filter(Musa_balbisiana == 1) %>% 
#   select(Orthogroup) %>% 
#   write.table("orthofinder/Orthogroups/VennDiag/Musa_balbisiana.OG.txt", col.names = F, sep = '\t', quote = F, row.names = F)



# Figure of main venn diagrams

# OG_Species = c("bedadeti","mazia","Musa_acuminata","Musa_balbisiana")

OG_list_VennDiag_TE <- list()

OG_list_VennDiag_TE[["EV (Bedadeti)"]] <- read.table("orthofinder/Orthogroups/VennDiag/bedadeti.OG.txt", header = F, stringsAsFactors = FALSE)$V1
OG_list_VennDiag_TE[["EV (Mazia)"]] <- read.table("orthofinder/Orthogroups/VennDiag/mazia.OG.txt", header = F, stringsAsFactors = FALSE)$V1

# OG_list_VennDiag_TE[["MA"]] <- read.table("orthofinder/Orthogroups/VennDiag/Musa_acuminata.OG.txt", header = F, stringsAsFactors = FALSE)$V1
# OG_list_VennDiag_TE[["MB"]] <- read.table("orthofinder/Orthogroups/VennDiag/Musa_balbisiana.OG.txt", header = F, stringsAsFactors = FALSE)$V1

```

# Plot Ven diagra 
```{r vendiag}
library(VennDiagram)

ggVennDiagram(x = OG_list_VennDiag_TE, 
              size = 4.3) +
  scale_fill_gradient(low="light grey", high = "red", limits = c(0, 18000)) +
  labs(fill="Count") +
  # ggtitle("Shared Orthogroups (OGs)") +
  theme(plot.title = element_text(hjust = 0.6),
        legend.position = "none")+
  scale_x_continuous(expand = expansion(mult = .2))
ggsave("orthofinder/Orthogroups/VennDiag/EV_AED25_MAB_VenDiag.tiff", width=3, height=2)
```



## summary statistics 
```{r summary_stat}
### statistics 
read.delim("orthofinder/Comparative_Genomics_Statistics/Statistics_Overall.tsv", sep = '\t', header = F) %>%
  slice(1:18) %>% 
  mutate(V1 = str_replace_all(V1, "Number", "#"),
         V1 = str_replace(V1, "Percentage", "%"),
         V1 = str_replace(V1, "orthogroup", "OG")) %>% 
  write.table("orthofinder/Comparative_Genomics_Statistics/Statistics_Overall.txt", sep = '\t', col.names = F, row.names = F, quote = F)
```

# Prepare data for duplication in species species OGs 
```{r dups_species_spec}
# Genome specific OGs
species_specific_OGs_all <-
  EV_AED25_TE.OG_GeneCount %>% 
  filter(
    bedadeti == 1,
    mazia==0 
    # Musa_acuminata == 0,
    # Musa_balbisiana ==0
  ) %>%
  select(Orthogroup) %>% 
  mutate(genome="EV_bedadeti") %>% 
  bind_rows(
    EV_AED25_TE.OG_GeneCount %>% 
      filter(
        bedadeti == 0,
        mazia==1
        # Musa_acuminata == 0,
        # Musa_balbisiana ==0
        
      ) %>%
      select(Orthogroup) %>% 
      mutate(genome="EV_mazia")
    # EV_AED25_TE.OG_GeneCount %>% 
    #   filter(
    #     bedadeti == 0,
    #     mazia==0, 
    #     Musa_acuminata == 1,
    #     Musa_balbisiana ==0
    #     
    #   ) %>%
    #   select(Orthogroup) %>% 
    #   mutate(genome="MA"),
    # EV_AED25_TE.OG_GeneCount %>% 
    #   filter(
    #     bedadeti == 0,
    #     mazia==0, 
    #     Musa_acuminata == 0,
    #     Musa_balbisiana ==1
    #     
    #   ) %>%
    #   select(Orthogroup) %>% 
    #   mutate(genome="MB")
    )


duplications_all <-
  read.delim("orthofinder/Gene_Duplication_Events/Duplications.tsv") 
```

### plot
```{r plot_dup_species_specific}
# duplications_all %>% 
species_specific_OGs_all %>% 
  # filter(genome == "EV_bedadeti") %>% 
  left_join(
    duplications_all
  ) %>% distinct() %>% 
  mutate(Species.Tree.Node = str_replace_na(Species.Tree.Node,"NA")) %>%
  filter(Species.Tree.Node != "NA") %>%
  select(Orthogroup, genome) %>% 
  distinct() %>% 
  group_by(genome) %>%
  summarise( duplications_OG = n()) %>%
  left_join(
    species_specific_OGs_all %>%
      group_by(genome) %>%
      summarise(total = n()) %>% 
      arrange(total) 
  ) %>%
  pivot_longer(!genome, names_to = "OG_category", values_to = "value") %>% 
  mutate(OG_category = str_replace(OG_category, "duplications_OG", "OGs with duplications event"),
         OG_category = str_replace(OG_category, "total", "Genome specific OGs")) %>% 
  # rename('OGs category' = OG_category) %>%
  ggplot(aes(x= factor(genome, levels = c("EV_bedadeti","EV_mazia")),
             y=value, fill= OG_category ))+
  geom_col( position = "dodge" )+
  scale_y_continuous(breaks =c(1139, 1927, 2477,6095))+
  scale_x_discrete(breaks =c("EV_bedadeti", "EV_mazia"),
                   labels = c("EV (Bedadeti)","EV (Mazia)"))+
  
  coord_flip()+
  # theme_bw() +
  labs(
    y =" Number of genome specifc orthogroups (OGs) or<br> OGs that went one or more duplications event",
    # x = "Genomes"
  )+
  theme(
    axis.title.x = element_markdown(size = 12, face = "bold"),
    axis.title.y = element_blank(),
    axis.line.x = element_line(),
    axis.line.y = element_line(),
    axis.text.y = element_markdown(size = 12),
    axis.text.x = element_markdown(size = 12),
    # axis.title.x
    panel.grid.major.x  = element_line(color ="#888888", size = 0.08),
    panel.background = element_rect(fill="#FFFFFF", color=NA),
    legend.title = element_blank(),
    legend.text = element_markdown(size = 12),
    legend.position = c(0.7,0.9)
  )
```


# Filter genes in duplications event 
```{r dup_speceis_specific_genes}
## bedadeti

bedadeti_unique_dup_gene1 <-
species_specific_OGs_all %>% 
  # filter(genome == "EV_bedadeti") %>% 
  left_join(
    duplications_all
  ) %>% 
  mutate(Species.Tree.Node = str_replace_na(Species.Tree.Node,"NA")) %>%
  filter(Species.Tree.Node != "NA") %>% 
  distinct() %>% 
  filter(genome == "EV_bedadeti" ) %>% 
  select(Orthogroup, Genes.1)  %>% 
  mutate(Genes.1 = str_remove_all(Genes.1,"bedadeti_"),
         Genes.1 = str_remove_all(Genes.1, "\\s\\w+\\s+\\w+_0.\\d+\\s+\\w+_0.\\d+"),
         Genes.1 = str_replace_all(Genes.1,"\\|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""),
         # ,
         # Genes.1 = str_replace_all(Genes.1,"\\s+\\d+|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""
         
         
         Genes.1 = str_replace_all(Genes.1,"\\d+\\|\\d+\\|\\d+\\|", ""),
         Genes.1 = str_replace_all(Genes.1,"_\\d+", "_"),
         Genes.1 = str_replace_all(Genes.1,"_.\\d+", "_"),
         Genes.1 = str_replace_all(Genes.1,"QI_", ""),
         Genes.1 = str_replace_all(Genes.1,"\\|\\d+", ""),
         Genes.1 = str_replace_all(Genes.1,"\\|--\\d+", "")
  ) %>%
    mutate(count = str_count(Genes.1,pattern = ","),
           count = as.numeric(count)) 
  # separate(Genes.1,sprintf("%s%02d", "A", 1:57), sep = "\\,") 

# total columns 
dup_count = bedadeti_unique_dup_gene1[,"count"] %>% max()

bedadeti_unique_dup_gene1 <-
bedadeti_unique_dup_gene1 %>%
  # mutate(count = str_count(Genes.1,pattern = ",")) %>%
  separate(Genes.1,sprintf("%s%02d", "A", 1:dup_count), sep = "\\,") %>%
  rotate_df() %>%  
  filter(str_detect(V2, "EV") | str_detect(V2, "^OG") ) 
  
# write.table("orthofinder/Gene_Duplication_Events/bedadeti_unique_dup_gene1.txt", col.names = F, row.names = F, quote = F, sep='\t')
# distinct()  %>% 
# mutate(genome="EV_beadeti") %>% 
bedadeti_unique_dup_gene2 <-
species_specific_OGs_all %>% 
  filter(genome == "EV_bedadeti") %>% 
  left_join(
    duplications_all
  ) %>% distinct() %>% 
  mutate(Species.Tree.Node = str_replace_na(Species.Tree.Node,"NA")) %>%
  filter(Species.Tree.Node != "NA") %>% 
  select(Orthogroup, Genes.2)  %>% 
  mutate(Genes.2 = str_remove_all(Genes.2,"bedadeti_"),
         Genes.2 = str_remove_all(Genes.2, "\\s\\w+\\s+\\w+_0.\\d+\\s+\\w+_0.\\d+"),
         Genes.2 = str_replace_all(Genes.2,"\\|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""),
         # ,
         # Genes.1 = str_replace_all(Genes.1,"\\s+\\d+|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""
         Genes.2 = str_replace_all(Genes.2,"\\d+\\|\\d+\\|\\d+\\|", ""),
         Genes.2 = str_replace_all(Genes.2,"_\\d+", "_"),
         Genes.2 = str_replace_all(Genes.2,"_.\\d+", "_"),
         Genes.2 = str_replace_all(Genes.2,"QI_", ""),
         Genes.2 = str_replace_all(Genes.2,"\\|\\d+", ""),
         Genes.2 = str_replace_all(Genes.2,"\\|--\\d+", "")
  ) %>%
    mutate(count = str_count(Genes.2,pattern = ","),
           count = as.numeric(count))
  # separate(Genes.1,sprintf("%s%02d", "A", 1:57), sep = "\\,") 

# total columns 
dup_count = bedadeti_unique_dup_gene2[,"count"] %>% max()

bedadeti_unique_dup_gene2 <-
bedadeti_unique_dup_gene2 %>%
  # mutate(count = str_count(Genes.1,pattern = ",")) %>%
  separate(Genes.2,sprintf("%s%02d", "A", 1:dup_count), sep = "\\,") %>%
  rotate_df() %>%  
  filter(str_detect(V2, "EV") | str_detect(V2, "^OG") ) 


  # write.table("orthofinder/Gene_Duplication_Events/bedadeti_unique_dup_gene2.txt", col.names = F, row.names = F, quote = F, sep = '\t')
# distinct()  %>% 
# mutate(genome="EV_beadeti") %>% 

## Mazia 
mazia_unique_dup_gene1 <- 
species_specific_OGs_all %>% 
  # filter(genome == "EV_bedadeti") %>% 
  left_join(
    duplications_all
  ) %>% 
  mutate(Species.Tree.Node = str_replace_na(Species.Tree.Node,"NA")) %>%
  filter(Species.Tree.Node != "NA") %>% 
  distinct() %>% 
  filter(genome == "EV_mazia" ) %>% 
  select(Orthogroup, Genes.1)  %>% 
  mutate(Genes.1 = str_remove_all(Genes.1,"mazia_"),
         Genes.1 = str_remove_all(Genes.1, "\\s\\w+\\s+\\w+_0.\\d+\\s+\\w+_0.\\d+"),
         Genes.1 = str_replace_all(Genes.1,"\\|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""),
         # ,
         # Genes.1 = str_replace_all(Genes.1,"\\s+\\d+|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""
         
         
         Genes.1 = str_replace_all(Genes.1,"\\d+\\|\\d+\\|\\d+\\|", ""),
         Genes.1 = str_replace_all(Genes.1,"_\\d+", "_"),
         Genes.1 = str_replace_all(Genes.1,"_.\\d+", "_"),
         Genes.1 = str_replace_all(Genes.1,"QI_", ""),
         Genes.1 = str_replace_all(Genes.1,"\\|\\d+", ""),
         Genes.1 = str_replace_all(Genes.1,"\\|--\\d+", "")
  ) %>%
    mutate(count = str_count(Genes.1,pattern = ","),
           count = as.numeric(count))
  # separate(Genes.1,sprintf("%s%02d", "A", 1:57), sep = "\\,") 

# total columns 
dup_count = mazia_unique_dup_gene1[,"count"] %>% max()

mazia_unique_dup_gene1 <-
mazia_unique_dup_gene1 %>%
  # mutate(count = str_count(Genes.1,pattern = ",")) %>%
  separate(Genes.1,sprintf("%s%02d", "A", 1:dup_count), sep = "\\,") %>%
  rotate_df() %>%  
  filter(str_detect(V2, "EV") | str_detect(V2, "^OG") )


  # write.table("orthofinder/Gene_Duplication_Events/mazia_unique_dup_gene1.txt", col.names = F, row.names = F, quote = F, sep = '\t')
# distinct()  %>% 
# mutate(genome="EV_beadeti") %>% 

mazia_unique_dup_gene2 <-
species_specific_OGs_all %>% 
  filter(genome == "EV_mazia") %>% 
  left_join(
    duplications_all
  ) %>% distinct() %>% 
  mutate(Species.Tree.Node = str_replace_na(Species.Tree.Node,"NA")) %>%
  filter(Species.Tree.Node != "NA") %>% 
  select(Orthogroup, Genes.2)  %>% 
  mutate(Genes.2 = str_remove_all(Genes.2,"mazia_"),
         Genes.2 = str_remove_all(Genes.2, "\\s\\w+\\s+\\w+_0.\\d+\\s+\\w+_0.\\d+"),
         Genes.2 = str_replace_all(Genes.2,"\\|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""),
         # ,
         # Genes.1 = str_replace_all(Genes.1,"\\s+\\d+|\\d+\\|\\d+\\|\\d+\\|\\d+\\|", ""
         Genes.2 = str_replace_all(Genes.2,"\\d+\\|\\d+\\|\\d+\\|", ""),
         Genes.2 = str_replace_all(Genes.2,"_\\d+", "_"),
         Genes.2 = str_replace_all(Genes.2,"_.\\d+", "_"),
         Genes.2 = str_replace_all(Genes.2,"QI_", ""),
         Genes.2 = str_replace_all(Genes.2,"\\|\\d+", ""),
         Genes.2 = str_replace_all(Genes.2,"\\|--\\d+", "")
  ) %>%
    mutate(count = str_count(Genes.2,pattern = ","),
           count = as.numeric(count))
  # separate(Genes.1,sprintf("%s%02d", "A", 1:57), sep = "\\,") 

# total columns 
dup_count = mazia_unique_dup_gene2[,"count"] %>% max()

mazia_unique_dup_gene2 <-
mazia_unique_dup_gene2 %>%
  # mutate(count = str_count(Genes.1,pattern = ",")) %>%
  separate(Genes.2,sprintf("%s%02d", "A", 1:dup_count), sep = "\\,") %>%
  rotate_df() %>%  
  filter(str_detect(V2, "EV") | str_detect(V2, "^OG") )

  # write.table("orthofinder/Gene_Duplication_Events/mazia_unique_dup_gene2.txt", col.names = F, row.names = F, quote = F, sep = '\t')
# distinct()  %>% 
# mutate(genome="EV_beadeti") %>% 

# EV_mazia_specific_OG_genes <- read.delim("orthofinder/orthofinder/Gene_Duplication_Events/mazia_unique_dup_gene2.txt", header = F, sep = '\t') %>%
#   mutate(row_names = str_extract(V1, "OG\\d+"),
#          V1 = str_remove(V1, "OG\\d+[\\s]"))
# 
# EV_mazia_specific_OG_genes %>% remove_rownames() %>% column_to_rownames(var = "row_names") %>% head()
# 
# add_rownames(EV_mazia_specific_OG_genes, var = EV_mazia_specific_OG_genes %>% 
#                mutate(row_names = str_extract(V1, "OG\\d+"),
#                       V1 = str_remove(V1, "OG\\d+[\\s]")) %>%
#                select(row_names)) 


## Read reformatted species specific OGs duplications 


# library(tidyverse)

## Bedadeti

orthofinder_dups_OG_gene_TE <- c()

# OG_Dup_Genes = c("bedadeti_unique_dup_gene1","bedadeti_unique_dup_gene2","mazia_unique_dup_gene1","mazia_unique_dup_gene2")
# OG_Dup_Genes = c("mazia_unique_dup_gene1")
# OG_Dup_Genes = c("bedadeti_unique_dup_gene2","mazia_unique_dup_gene2")

# OG_Dup_Genes = c(bedadeti_unique_dup_gene2,mazia_unique_dup_gene2)


  unique_dup_OG <- mazia_unique_dup_gene2

  
  for (i in 1:ncol(unique_dup_OG)) {
    
    orthofinder_dups <- 
      unique_dup_OG[i] 
    
    
    colnames(orthofinder_dups) = c("V1")
    
    orthofinder_dups_OG_gene_TE <-
      orthofinder_dups %>% 
      mutate(OGs = str_extract(V1,"^OG\\d+"),
             OGs = str_replace_na(OGs, OGs)
             # OGs_from = case_when(str_detect(V1,"^EVMZ")~ "mazia")
             ) %>%
      filter(!str_detect(V1,"^OG"), V1 !="") %>%
      rbind(orthofinder_dups_OG_gene_TE)
  }
  

  
    
    unique_dup_OG <- bedadeti_unique_dup_gene2


  
  for (i in 1:ncol(unique_dup_OG)) {
    
    orthofinder_dups <- 
      unique_dup_OG[i] 
    
    
    colnames(orthofinder_dups) = c("V1")
    
    orthofinder_dups_OG_gene_TE <-
      orthofinder_dups %>% 
      mutate(OGs = str_extract(V1,"^OG\\d+"),
             OGs = str_replace_na(OGs, OGs)
             # OGs_from = case_when(str_detect(V1,"^EVMZ")~ "mazia")
             ) %>%
      filter(!str_detect(V1,"^OG"), V1 !="") %>%
      rbind(orthofinder_dups_OG_gene_TE)
  }
    
    
    # label genomes 
    bed_maz_orthofinder_dups_OG_gene_TE <- 
    orthofinder_dups_OG_gene_TE %>% 
      mutate( OGs_from = case_when(str_detect(V1,"EVMZ") ~ "mazia",
                                   str_detect(V1,"EVBD") ~ "bedadeti")) %>%
      rename(seq.name = V1) 
    
    # remove row names
                
      row.names(bed_maz_orthofinder_dups_OG_gene_TE) <- NULL
      
      bed_maz_orthofinder_dups_OG_gene_TE <- 
      bed_maz_orthofinder_dups_OG_gene_TE %>%
        distinct()
      
      bed_maz_orthofinder_dups_OG_gene_TE %>% distinct(OGs_from)
    
    
  # orthofinder_dups_OG_gene_TE %>% 
  #   distinct() %>% 
  #   # select(OGs) %>% distinct() %>% nrow()
  #   write.table(paste0("orthofinder/orthofinder/Gene_Duplication_Events/", OG, ".1.txt"),sep = '\t',col.names = F,row.names = F,quote = F)
# }


# maz_bed_dups_OG_gene <- 
#   orthofinder_dups_OG_gene_TE %>%
#   mutate(OGs_from = str_remove_all(OGs_from,"_unique_dup_gene\\d"))

# orthofinder_dups_OG_gene_TE %>% 
#   select(-OGs) %>%
#   distinct() %>%
#   group_by(OGs_from) %>% 
#   summarise(count_dup_OG_genes=n())
# unique_dup_OG %>%
#   head()
# 
# 
# orthofinder_dups_OG_gene_TE %>%
#   head()


bed_maz_orthofinder_dups_OG_gene_TE %>% 
  # rename(seq.name=V1) %>%
  distinct() %>%
  group_by(OGs_from) %>% 
  summarise(count_dup_OG_genes=n())



bed_maz_orthofinder_dups_OG_gene_TE %>% tail()

```

# Parse assigned OG genes 
```{r assigned OG_genes}

### final script: parse_OG_gene_list 


library(tidyverse)

## read OGs containing gene list

orthogroups_TE_gene_list <- read.delim("orthofinder/Orthogroups/Orthogroups.tsv", header = T)  

orthogroups_TE_gene_list %>% colnames()

# read gene list count table for each OGs 
OG_GeneCount_unfiltered_TE <- read.delim("orthofinder/Orthogroups/Orthogroups.GeneCount.tsv", sep = '\t', header = T) %>% 
  dplyr::rename(
    
    # "MA" = Musa_acuminata,
    # "MB" = Musa_balbisiana,
    "EV_bed" = bedadeti,
    "EV_maz" = mazia) #%>%


# get the maximum number of genes contained in OGs for each genome 

orthogroups_TE_gene_list_max_genes <-
  # orthogroups_TE_gene_list %>%
  OG_GeneCount_unfiltered_TE %>%
  # select(-Ensete_glaucum,-bedadeti,-mazia,
  #        -Musa_acuminata,-Musa_balbisiana,-Musa_schizocarpa) %>%
  # group_by(Orthogroup) %>%
  summarise(
    #Ensete_glaucum = max(EG),
    bedadeti = max(EV_bed),
    mazia = max(EV_maz),
    # Musa_acuminata  = max(MA),
    # Musa_balbisiana = max(MB)
    # Musa_schizocarpa = max(MS)
  ) 



Species_List = c(
                 "bedadeti", "mazia")

all_assigned_OGsgenes_TE <- c()


for (Each_Species in Species_List) {
  # set maximum column number for the species 
  col_range = orthogroups_TE_gene_list_max_genes %>%
    dplyr::select(Each_Species) 
  
  
  col_range = 1:(col_range[1,1]+1)
  
  # split to genes to columns and 
  # Species_OGs <-
  all_assigned_OGsgenes_TE <-
    orthogroups_TE_gene_list %>%
    dplyr::select(Orthogroup, Each_Species) %>%
    separate(Each_Species,"seq.name", sep = "\\,")     %>%
    ## create Column IDS to combine and assign spitted genes into their respective genomes
    mutate(genome = Each_Species) %>%
    mutate(seq.name  = str_replace_na(seq.name, "") ) %>%
    dplyr::filter(seq.name !="" ) %>%
    rbind(all_assigned_OGsgenes_TE)
  
  
  # Col_Names_List <- 
  #   Species_OGs %>%
  #   dplyr::select(-Orthogroup) %>% 
  #   colnames() 
  
  # for (Each_Col_Name in Col_Names_List) {
  #   
  #   all_assigned_OGsgenes_TE <- 
  #     Species_OGs %>%
  #     dplyr::select(Orthogroup, Each_Col_Name) %>%
  #     dplyr::rename(pred_genes = Each_Col_Name) %>%
  #     mutate(species = Each_Species) %>%
  #     rbind(all_assigned_OGsgenes_TE)
  # }
  # 
  # all_assigned_OGsgenes_TE <-
  #   all_assigned_OGsgenes_TE %>% 
  #   mutate(pred_genes  = str_replace_na(pred_genes, "NA") ) %>%
  #   dplyr::filter(pred_genes != "NA", pred_genes !="" ) 
  
}


all_assigned_OGsgenes_TE %>% distinct(genome)


```

# Parse Unassigned OG genes 

```{r }
unassigned_prot_TE <- "orthofinder/Orthogroups/Orthogroups_UnassignedGenes.tsv"


Orthogroups_UnassignedGenes_TE <- read.delim(unassigned_prot_TE, header = T, sep = '\t')

Genomes_Lists=c("bedadeti",
                "mazia")

# Orthogroups_UnassignedGenes_TE %

# Orthogroups_UnassignedGenes_TE_EV <- Orthogroups_UnassignedGenes_TE_list

Orthogroups_UnassignedGenes_TE_EV <- c()

# for (Each_Genome in Genomes_Lists) {
#   Orthogroups_UnassignedGenes_TE_list <- 
#     Orthogroups_UnassignedGenes_TE %>% 
#     select(Orthogroup, Each_Genome) %>% 
#     distinct() %>%
#     # filter(Name !="") %>% 
#     # rename(seq.name=Ensete_glaucum) %>%
#     mutate(
#       # seq.name = str_extract(Name,"\\w+[^\\s]+"),
#       genome = Each_Genome) %>%
#     rename(seq.name = Each_Genome) %>%
#     rbind(Orthogroups_UnassignedGenes_TE_list)
# }


### updated 
Orthogroups_UnassignedGenes_TE_EV  <- c()

for (Each_Genome in Genomes_Lists) {
  Orthogroups_UnassignedGenes_TE_EV <- 
    Orthogroups_UnassignedGenes_TE %>% 
    select(Orthogroup, Each_Genome) %>% 
    distinct() %>% 
    # filter(Name !="") %>% 
    # rename(seq.name=Ensete_glaucum) %>%
    mutate(
      # seq.name = str_extract(Name,"\\w+[^\\s]+"),
      genome = Each_Genome) %>% 
    rename(seq.name = Each_Genome) %>% 
    filter(seq.name != "") %>% 
    rbind(Orthogroups_UnassignedGenes_TE_EV)
  
}


Orthogroups_UnassignedGenes_TE_EV %>% 
  filter(seq.name !="") %>% 
  group_by(genome) %>%
  summarise(count = n()) %>%
  
  ggplot(aes(x= factor(genome, levels = c("bedadeti","mazia")),
             y=count ))+
  geom_col( position = "dodge" )+
  # scale_y_continuous(breaks =c(85,  320, 1177))+
  scale_x_discrete(breaks =c("bedadeti","mazia"),
                   labels = c("EV (Bedadeti)","EV (Mazia)"))+
  # scale_y_continuous(breaks =c(1380,2398, 3311, 5955, 6418))+
  
  
  coord_flip()+
  # theme_bw() +
  labs(
    y  = " Number of unassigned orthogroups or genes",
    # x = "Genomes"
  )+
  theme(
    axis.title.x = element_markdown(size = 12, face = "bold"),
    axis.title.y = element_blank(),
    axis.line.x = element_line(),
    axis.line.y = element_line(),
    axis.text.y = element_markdown(size = 12),
    axis.text.x = element_markdown(size = 10),
    # axis.title.x
    panel.grid.major.x  = element_line(color ="#888888", size = 0.08),
    panel.background = element_rect(fill="#FFFFFF", color=NA),
    legend.title = element_blank(),
    legend.text = element_markdown(size = 12)
    # legend.position = c(0.7,0.9)
  )

ggsave("orthofinder/Orthogroups/VennDiag/Unassigned_OG_genes.tiff", width=7, height=7)
```





# Orthogroups by type: Multi-single-unique-unassinged 
## prepare data
```{r ortho_type}
unfiltered_EV_AED25_TE.OG_GeneCount <-
  read.delim("orthofinder/Orthogroups/Orthogroups.GeneCount.tsv", sep = '\t', header = T) %>%
  bind_rows(
    read.delim("orthofinder/Orthogroups/Orthogroups_UnassignedGenes.tsv") %>% 
      mutate(
        
        mazia = case_when(str_detect(mazia,"\\w+") ~ 1,
                                             TRUE ~ 0),
        bedadeti = case_when(str_detect(bedadeti, "\\w+") ~ 1,
                                                TRUE ~ 0)
        # Musa_acuminata = case_when(str_detect(Musa_acuminata, "\\w+") ~ 1,
        #                            TRUE ~ 0),
        # Musa_balbisiana = case_when(str_detect(Musa_balbisiana, "\\w+") ~ 1,
        #                             TRUE ~ 0)
      ))


single_copy_ortholgos_TE <- 
  unfiltered_EV_AED25_TE.OG_GeneCount %>%
  filter(
    
    # Musa_acuminata == 1,
    # Musa_balbisiana == 1,
    bedadeti == 1,
    mazia ==1
  ) %>%
  mutate (
    single_copy_OGs = "yes"
  ) %>%
  select(-Total)

# summarise(
#   Ensete_glaucum = sum(Ensete_glaucum), 
#   Musa_acuminata = sum(Musa_acuminata),
#   Musa_balbisiana = sum(Musa_balbisiana),
#   Musa_schizocarpa = sum(Musa_schizocarpa),
#   bedadeti = sum(bedadeti),
#   mazia = sum(mazia)
# ) %>% 
# pivot_longer( cols = Ensete_glaucum:mazia,
#               values_to = "value")%>%
# mutate(Orthogroup = "Single-copy orthologs")

unassigned_OGs_TE <-
  unfiltered_EV_AED25_TE.OG_GeneCount %>%
  # left_join(
  #   single_copy_ortholgos_TE
  # ) %>% 
  # mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
  # filter(single_copy_OGs == "NA") %>%
  filter(
    
    # Musa_acuminata == 1,
    # Musa_balbisiana == 0,
    bedadeti == 1,
    mazia ==0
  ) %>% 
  rbind(
    
    unfiltered_EV_AED25_TE.OG_GeneCount %>%
      # left_join(
      #   single_copy_ortholgos_TE
      # ) %>% 
      # mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
      # filter(single_copy_OGs == "NA") %>%
      filter(
        
        # Musa_acuminata == 0,
        # Musa_balbisiana == 1,
        bedadeti == 0,
        mazia ==1
      )#,
    # unfiltered_EV_AED25_TE.OG_GeneCount %>%
    #   left_join(
    #     single_copy_ortholgos_TE
    #   ) %>% 
    #   mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
    #   filter(single_copy_OGs == "NA") %>%
    #   filter(
    #     
    #     Musa_acuminata == 0,
    #     Musa_balbisiana == 0,
    #     bedadeti == 1,
    #     mazia ==0
    #   ),
    # unfiltered_EV_AED25_TE.OG_GeneCount %>%
    #   left_join(
    #     single_copy_ortholgos_TE
    #   ) %>% 
    #   mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
    #   filter(single_copy_OGs == "NA") %>%
    #   filter(
    #     
    #     Musa_acuminata == 0,
    #     Musa_balbisiana == 0,
    #     bedadeti == 0,
    #     mazia ==1
    #   )
  ) %>% 
  mutate(unassigned_OG = "yes") %>%
  select (-Total)


single_copy_ortholgos_TE %>% head()

unassigned_OGs_TE %>% head()
unassigned_OGs_TE %>% tail()


unique_OGs_TE <-
  unfiltered_EV_AED25_TE.OG_GeneCount %>%
  left_join(
    single_copy_ortholgos_TE
  ) %>% 
  mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
  filter(single_copy_OGs == "NA") %>%
  filter(
    
    # Musa_acuminata > 1,
    # Musa_balbisiana == 0,
    bedadeti > 1,
    mazia ==0
  ) %>% 
  rbind(
    
    unfiltered_EV_AED25_TE.OG_GeneCount %>%
      left_join(
        single_copy_ortholgos_TE
      ) %>% 
      mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
      filter(single_copy_OGs == "NA") %>%
      filter(
        
        # Musa_acuminata == 0,
        # Musa_balbisiana > 1,
        bedadeti == 0,
        mazia > 1
      )#,
    
    # unfiltered_EV_AED25_TE.OG_GeneCount %>%
    #   left_join(
    #     single_copy_ortholgos_TE
    #   ) %>% 
    #   mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
    #   filter(single_copy_OGs == "NA") %>%
    #   filter(
    #     
    #     Musa_acuminata == 0,
    #     Musa_balbisiana == 0,
    #     bedadeti > 1,
    #     mazia ==0
    #   ),
    # unfiltered_EV_AED25_TE.OG_GeneCount %>%
    #   left_join(
    #     single_copy_ortholgos_TE
    #   ) %>% 
    #   mutate(single_copy_OGs = str_replace_na(single_copy_OGs,"NA")) %>%
    #   filter(single_copy_OGs == "NA") %>%
    #   filter(
    #     
    #     Musa_acuminata == 0,
    #     Musa_balbisiana == 0,
    #     bedadeti == 0,
    #     mazia > 1
    #   )
  ) %>% 
  mutate(unique_OGs = "yes") %>%
  select(-single_copy_OGs,-Total) 

## multi-copy OGs 

multi_copy_OGs_TE <-
  unfiltered_EV_AED25_TE.OG_GeneCount %>% 
  left_join(
    single_copy_ortholgos_TE
  ) %>%
  left_join(
    unique_OGs_TE
  ) %>% 
  left_join(
    unassigned_OGs_TE
  ) %>% 
  mutate(
    single_copy_OGs = str_replace_na(single_copy_OGs,"NA"),
    unique_OGs = str_replace_na(unique_OGs,"NA"),
    unassigned_OG = str_replace_na(unassigned_OG,"NA")
  ) %>% 
  filter(
    single_copy_OGs == "NA" ,
    unique_OGs == 'NA' ,
    unassigned_OG == "NA"
  ) %>%
  distinct() %>%
  select(-single_copy_OGs, -unique_OGs, -unassigned_OG, -Total)


# multi_copy_OGs %>% head()

# unique_OGs
# single_copy_ortholgos_TE
# unassigned_OGs

multi_single_unique_unassigned_OG_TE_summary <-
  multi_copy_OGs_TE %>%
  summarise(
    # Ensete_glaucum = sum(Ensete_glaucum),
    # Musa_acuminata = sum(Musa_acuminata),
    # Musa_balbisiana = sum(Musa_balbisiana),
    # Musa_schizocarpa = sum(Musa_schizocarpa),
    bedadeti = sum(bedadeti),
    mazia = sum(mazia)
  ) %>% 
  pivot_longer( cols = bedadeti:mazia,
                values_to = "value")%>%
  mutate(Orthogroup = "Multi-copy orthologs") %>%
  rbind(
    single_copy_ortholgos_TE %>%
      summarise(
        # Ensete_glaucum = sum(Ensete_glaucum),
        # Musa_acuminata = sum(Musa_acuminata),
        # Musa_balbisiana = sum(Musa_balbisiana),
        # Musa_schizocarpa = sum(Musa_schizocarpa),
        bedadeti = sum(bedadeti),
        mazia = sum(mazia)
      ) %>%
      pivot_longer( cols = bedadeti:mazia,
                    values_to = "value")%>%
      mutate(Orthogroup = "Single-copy orthologs"),
    unique_OGs_TE %>%
      summarise(
        # Ensete_glaucum = sum(Ensete_glaucum),
        # Musa_acuminata = sum(Musa_acuminata),
        # Musa_balbisiana = sum(Musa_balbisiana),
        # Musa_schizocarpa = sum(Musa_schizocarpa),
        bedadeti = sum(bedadeti),
        mazia = sum(mazia)
      ) %>%
      pivot_longer( cols = bedadeti:mazia,
                    values_to = "value")%>%
      mutate(Orthogroup = "Unique"),
    unassigned_OGs_TE %>%
      summarise(
        # Ensete_glaucum = sum(Ensete_glaucum),
        # Musa_acuminata = sum(Musa_acuminata),
        # Musa_balbisiana = sum(Musa_balbisiana),
        # Musa_schizocarpa = sum(Musa_schizocarpa),
        bedadeti = sum(bedadeti),
        mazia = sum(mazia)
      ) %>%
      pivot_longer( cols = bedadeti:mazia,
                    values_to = "value")%>%
      mutate(Orthogroup = "Unassigned")
  ) 
```


## Assigned_unassigned_combined
```{r all OGs}

assigned_unassigned_OGs_TE_EV <-
  multi_copy_OGs_TE %>% 
  mutate (
    bedadeti = case_when(bedadeti > 1 ~ "bedadeti" ),
    mazia= case_when(mazia > 1 ~ "mazia"),
    # Musa_acuminata= case_when(Musa_acuminata > 1 ~ "musa_ac"),
    # Musa_balbisiana= case_when(Musa_balbisiana > 1 ~ "musa_ba")
    ) %>%
  bind_rows(
    unique_OGs_TE  %>%
      mutate (
        bedadeti = case_when(bedadeti >=1 ~ "bedadeti" ),
        mazia= case_when(mazia >=1 ~ "mazia"),
        # Musa_acuminata= case_when(Musa_acuminata >=1 ~ "musa_ac"),
        # Musa_balbisiana= case_when(Musa_balbisiana >=1 ~ "musa_ba")
        ), 
    unassigned_OGs_TE %>%
      mutate (
        bedadeti = case_when(bedadeti ==1 ~ "bedadeti" ),
        mazia= case_when(mazia ==1 ~ "mazia"),
        # Musa_acuminata= case_when(Musa_acuminata ==1 ~ "musa_ac"),
        # Musa_balbisiana= case_when(Musa_balbisiana ==1 ~ "musa_ba")
        ),
    single_copy_ortholgos_TE %>%
      mutate (
        bedadeti = case_when(bedadeti ==1 ~ "bedadeti" ),
        mazia= case_when(mazia ==1 ~ "mazia"),
        # Musa_acuminata= case_when(Musa_acuminata ==1 ~ "musa_ac"),
        # Musa_balbisiana= case_when(Musa_balbisiana ==1 ~ "musa_ba")
        )
    ) 

assigned_unassigned_OGs_TE_EV.1 <- 
  assigned_unassigned_OGs_TE_EV %>% 
  pivot_longer( cols = bedadeti:mazia,
                values_to = "value") %>%
  select(Orthogroup,name,value) %>%
  mutate(value = str_replace_na(value, "NA")) %>%
  filter(value != "NA") %>%
  distinct()

# assigned_unassigned_OGs_EV_MAB_v1 %>%
#   rename(genome = name) %>%
#   left_join(
#     
#     Orthogroups_UnassignedGenes_TE_EV %>% 
#       head() %>%
#       bind_rows(
#         all_assigned_OGsgenes_TE) %>% 
#       # rename(#seq.name = pred_genes, 
#       #        genome = species)) %>%
#       distinct()) %>% tail()


# EV_AED25_TE.OG_GeneCount %>%
#   filter(Orthogroup == "OG0016389")

assigned_unassigned_OGs_genes_TE_EV <-
  read.delim("orthofinder/Orthogroups/Orthogroups.txt", sep = ' ', header = F) %>%  
  pivot_longer( cols = V2:V459,
                values_to = "seq.name") %>%
  mutate(V1 = str_remove(V1,":"),
         genome = case_when (str_detect(seq.name,"EVBD") ~ "bedadeti",
                             str_detect(seq.name,"EVMZ") ~ "mazia",
                             # str_detect(seq.name, "Eg\\d+") ~ "ensete_gl",
                             # str_detect(seq.name, "Mba") ~ "musa_ba",
                             # str_detect(seq.name, "Macma") ~ "musa_ac",
                             # str_detect(seq.name) ~ "musa_sc"
         )) %>%
  rename(Orthogroup = V1) %>% 
  select(Orthogroup, seq.name,genome) %>%
  mutate(genome = str_replace_na(genome, "NA")) %>%
  filter(genome != "NA") %>%
  
  # bind unassigned OGs
  
  bind_rows(Orthogroups_UnassignedGenes_TE_EV)
  


assigned_unassigned_OGs_genes_TE_EV %>%
  tail()
```



## plot

```{r plot}
multi_single_unique_unassigned_OG_TE_summary.1 <- 
  multi_single_unique_unassigned_OG_TE_summary %>%
  # filter(name == "mazia")
  mutate(name = str_replace_all(name,c("bedadeti" = "*E. ventricosum* (Bedadeti)",
                                       "mazia" = "*E. ventricosum* (Mazia)"))) 



multi_single_unique_unassigned_OG_TE_summary.1$name <- factor(multi_single_unique_unassigned_OG_TE_summary.1$name, 
                                                            levels = c("*E. ventricosum* (Mazia)", 
                                                                       "*E. ventricosum* (Bedadeti)"))

multi_single_unique_unassigned_OG_TE_summary.1 %>%
  ggplot(aes(name,value, fill = Orthogroup)) +
  geom_col()+
  # scale_y_continuous(breaks =c(0, 2239, 12890, 26000,39000))+
  # scale_x_discrete(breaks =c( "MB", "MA", "EV_bedadeti", "EV_mazia"),
  #                  labels = c("MB","MA",  "EV (Bedadeti)","EV (Mazia)"))+
  
  coord_flip()+
  # theme_bw() +
  labs(
    y =" Number of genes",
    # x = "Genomes"
  )+
  theme(
    axis.title.x = element_markdown(size = 12, face = "bold"),
    axis.title.y = element_blank(),
    axis.line.x = element_line(),
    axis.line.y = element_line(),
    axis.text.y = element_markdown(size = 12),
    axis.text.x = element_markdown(size = 12),
    # axis.title.x
    panel.grid.major.x  = element_line(color ="#888888", size = 0.08),
    panel.background = element_rect(fill="#FFFFFF", color=NA),
    legend.title = element_blank(),
    # legend.text = element_markdown(size = 12),
    legend.text = element_markdown(size=10.8),
    legend.key.size = unit(0.2,"cm"), 
    legend.position = "top"
  )

ggsave("orthofinderOG_EV_TE_summary.tiff", width=7, height=4)

```

```{r save}
save.image(file = "orthofinder/EV_AED25_MAB/OrthoFinder.EV.AED25.TE.RData")
# load(file = "orthofinder/EV_AED25_MAB/OrthoFinder.EV.AED25.MAB.RData")

```